src/mscorlib/shared/System/Buffers/OwnedMemory.cs

```diff
@@ -0,0 +1,65 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime;
+using System.Runtime.CompilerServices;
+
+namespace System.Buffers
+{
+    public abstract class OwnedMemory<T> : IDisposable, IRetainable
+    {
+        protected OwnedMemory() { }
+
+        public abstract int Length { get; }
+
+        public abstract Span<T> AsSpan(int index, int length);
+
+        public virtual Span<T> AsSpan()
+        {
+            if (IsDisposed) ThrowHelper.ThrowObjectDisposedException(nameof(OwnedMemory<T>), ExceptionResource.Memory_ThrowIfDisposed);
+            return AsSpan(0, Length);
+        }
+
+        public Memory<T> Memory
+        {
+            get 
+            {
+                if (IsDisposed) ThrowHelper.ThrowObjectDisposedException(nameof(OwnedMemory<T>), ExceptionResource.Memory_ThrowIfDisposed);
+                return new Memory<T>(this, 0, Length);
+            }
+        }
+
+        public ReadOnlyMemory<T> ReadOnlyMemory
+        {
+            get 
+            {
+                if (IsDisposed) ThrowHelper.ThrowObjectDisposedException(nameof(OwnedMemory<T>), ExceptionResource.Memory_ThrowIfDisposed);
+                return new ReadOnlyMemory<T>(this, 0, Length);
+            }
+        }
+
+        public abstract MemoryHandle Pin();
+
+        protected internal abstract bool TryGetArray(out ArraySegment<T> arraySegment);
+
+        public void Dispose()
+        {
+            if (IsRetained) ThrowHelper.ThrowInvalidOperationException(ExceptionResource.Memory_OutstandingReferences);
+            Dispose(true);
+        }
+
+        protected abstract void Dispose(bool disposing);
+
+        protected abstract bool IsRetained { get; }
+
+        public abstract bool IsDisposed { get; }
+
+        public abstract void Retain();
+
+        public abstract bool Release();
+
+        internal static readonly T[] EmptyArray = new T[0];
```

Comment:
Why is this needed rather than `Array.Empty<T>()`?

Reply:
From: https://github.com/dotnet/corefxlab/pull/1569#discussion_r118106479
> We don't have it in NS 1.1, at least I don't see it.

I will use `Array.Empty<T>()` here but the impl in System.Memory will need to have this EmptyArray property.